### PR TITLE
Added torch model and training pipeline

### PR DESCRIPTION
**Pull request by LucaMalagutti**
This pull request adds a Pytorch version of the TIMNET architecture together with extendable training and evaluation pipelines that follow the already existing code relying on TF/Keras.

This addition could be useful to users that are more familiar with the torch ecosystem.

### CODE REVIEW

**Comment by hwang9u** [comment]
Nice work!!!

**Comment by whh07141** [comment]
Hello author, I reproduced your code, but the test results are very poor and the loss on the verification set does not decrease. Could you please tell me about your parameter settings?
